### ISSUE TITLE
Index fails to increment in for loop

### ISSUE THREAD

**Issue by pielover928** [ItsDeltin/Overwatch-Script-To-Workshop#63]
When I made this loop, it didn't increment i and got stuck in an infinite loop:

```for(define i = 0; i < someObj.numOptions; i++) {
				BigMessage(i); //psuedocode
}```

The above repeatedly says "0". Changing it to this works:

```for(define i = 0; i < someObj.numOptions; i++) {
				BigMessage(i); //psuedocode
				i++;
}```


However, changing to this broke it again:
```for(define i = 0; i < someObj.numOptions; i++) {
				BigMessage(i+1); //psuedocode
				i++;
}```

**Comment by ItsDeltin** [comment]
The following worked for me, can you post the full script?
```
rule: "Test"
Event.OngoingPlayer
if (IsButtonHeld(EventPlayer(), Button.Interact))
{
    myStruct someObj = new myStruct();
    for(define i = 0; i < someObj.numOptions; i++)
    {
        SmallMessage(EventPlayer(), i);
        MinWait(); // There is a workshop glitch where SmallMessage will display the wrong value if the value is changed immediately after calling SmallMessage.
    }
}

struct myStruct
{
    define numOptions = 3;
}
```

**Comment by pielover928** [comment]
```
struct Menu{
    define numOptions;
    define optionTextIds = []; //[0] = option 0 and so on

    define positions = [];
    define angle;


    define hovered; // 0 to numOptions
    define clicked; //Same

    method Create(){
        angle = numOptions*10;
        for(define i = 0; i < numOptions; i++){
            positions[i] = EyePosition(EventPlayer()) + 2*DirectionFromAngles(HorizontalFacingAngleOf(EventPlayer()) + (angle/2) - (i*10) - 5, 0);
            CreateIcon(EventPlayer(), positions[i], Icon.Circle, IconRev.None, Color.Blue, false);
            optionTextIds[i] = LastTextID();
        }
    }

    method Update(){

    }

    method Destroy(){
        foreach(define id in optionTextIds){
            DestroyInWorldText(id);
        }
    }
}

define globalvar planes;
define globalvar spheres;
define globalvar rects;

define globalvar hostInCreateMode;
define globalvar creationType;
Menu globalvar createMenu M;

rule: "Host Enters Creating Mode"
Event.OngoingPlayer
if(EventPlayer() == HostPlayer() & !hostInCreateMode & IsCommunicating(EventPlayer(), Communication.VoiceLineUp)){

    hostInCreateMode = true;
    creationType = -1;

    SetMoveSpeed(EventPlayer(),0);
    
    createMenu.numOptions = 3;
    createMenu.clicked = 0;
    createMenu.hovered = 0;
    createMenu.Create();
}

rule: "Host Hovers Over Menu Option"
Event.OngoingPlayer
if(EventPlayer() == HostPlayer() & hostInCreateMode & createMenu.clicked == 0){
    for(define i = 0; i < createMenu.numOptions; i++){
        BigMessage(EventPlayer(),EyePosition(EventPlayer()) + 2*FacingDirectionOf(EventPlayer()));
        BigMessage(EventPlayer(), createMenu.positions[i]);
        if(DistanceBetween(EyePosition(EventPlayer()) + 2*FacingDirectionOf(EventPlayer()), createMenu.positions[i]) < 1){
            if(createMenu.hovered != i+1){
                createMenu.hovered = i+1;            
            }
            //BigMessage(EventPlayer(), i);
            break;
            
        }
        i += 1;
    }
    if(createMenu.clicked == 0){
        Loop;
    }
}
```

**Comment by ItsDeltin** [comment]
`Loop;` and `break;` is not valid. That is supposed to show a syntax error, which will be fixed in 1.4. 
For me, `i+=1` makes the loop count 2 at a time and removing it makes it work as expected.

**Comment by pielover928** [comment]
0.4.0.2
